src/mlpack/tests/ann_layer_test.cpp

```diff
@@ -2168,6 +2168,80 @@ TEST_CASE("BilinearInterpolationLayerParametersTest", "[ANNLayerTest]")
   REQUIRE(layer1.InDepth() == layer2.InDepth());
 }
 
+/*
+ * Simple test for the BicubicInterpolation layer
+ */
+TEST_CASE("SimpleBicubicInterpolationLayerTest", "[ANNLayerTest]")
+{
+  // Tested output against  torch.nn.Upsample(mode="bicubic")
+  // Test case with square input with rectangular output
+  arma::mat input, output, unzoomedOutput, expectedOutput;
+  size_t inRowSize = 2;
+  size_t inColSize = 2;
+  size_t outRowSize = 5;
+  size_t outColSize = 7;
+  size_t depth = 1;
+  input.zeros(inRowSize * inColSize * depth, 1);
+
+  input << 10 << 20 << arma::endr
+        << 30 << 40 << arma::endr;
+  input.reshape(4, 1);
+  BicubicInterpolation<> layer(inRowSize, inColSize, outRowSize, outColSize, depth);
+
+  expectedOutput << 6.68803935860  <<  7.33308309038 <<  9.69733236152 << 12.79500000000 << 15.89266763848 << 18.25691690962 << 18.90196064140 << arma::endr
+                 << 10.53303935860 << 11.17808309038 << 13.54233236152 << 16.64000000000 << 19.73766763848 << 22.10191690962 << 22.74696064140 << arma::endr
+                 << 18.89303935860 << 19.53808309038 << 21.90233236152 << 25.00000000000 << 28.09766763848 << 30.46191690962 << 31.10696064140 << arma::endr
+                 << 27.25303935860 << 27.89808309038 << 30.26233236152 << 33.36000000000 << 36.45766763848 << 38.82191690962 << 39.46696064140 << arma::endr
+                 << 31.09803935860 << 31.74308309038 << 34.10733236152 << 37.20500000000 << 40.30266763848 << 42.66691690962 << 43.31196064140 << arma::endr;
```

Comment:
I think we have to break this down a little bit to make it fit in the 80 characters limit - https://github.com/mlpack/mlpack/wiki/DesignGuidelines#line-length-and-wrapping